src/d3a/d3a_core/sim_results/stats.py

```diff
@@ -188,6 +147,51 @@ def update_cumulative_bills(self, area):
             self.cumulative_bills_results[area.uuid]["penalties"] += penalty_cost
             self.cumulative_bills_results[area.uuid]["total"] += total
 
+
+class MarketEnergyBills:
+    def __init__(self, is_spot_market=True):
+        self.is_spot_market = is_spot_market
+        self.bills_results = {}
+        self.bills_redis_results = {}
+        self.market_fees = {}
+
+    @classmethod
+    def _store_bought_trade(cls, result_dict, trade):
+        # Division by 100 to convert cents to Euros
+        fee_price = trade.fee_price / 100. if trade.fee_price is not None else 0.
+        result_dict['bought'] += trade.offer.energy
+        result_dict['spent'] += trade.offer.price / 100. - fee_price
+        result_dict['total_energy'] += trade.offer.energy
+        result_dict['total_cost'] += trade.offer.price / 100.
+        result_dict['market_fee'] += fee_price
+
+    @classmethod
+    def _store_sold_trade(cls, result_dict, trade):
+        # Division by 100 to convert cents to Euros
+        fee_price = trade.fee_price if trade.fee_price is not None else 0.
+        result_dict['sold'] += trade.offer.energy
+        result_dict['earned'] += (trade.offer.price - fee_price) / 100.
+        result_dict['total_energy'] -= trade.offer.energy
+        result_dict['total_cost'] -= (trade.offer.price - fee_price) / 100.
+
+    def _default_area_dict(self, area):
```

Comment:
I think it could also be a class method

Reply:
Indeed no need for self here, converting this to a class method.